grails-app/controllers/au/org/ala/biocollect/merit/SiteController.groovy

```diff
@@ -338,8 +346,15 @@ class SiteController {
             }
         }
 
-        if (!result)
+        if (!result) {
             result = siteService.updateRaw(id, values)
+            if(postBody.pActivityId){
```

Comment:
Check whether user can update the Project Activity Id. 

This can be done by retrieving projectId (instead of relying on supplied projecttId) from project activity id and checking user permission against it.